File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.cpp
class Solution {
public:
    string smallestPalindrome(string s) {
        int half=s.length()/2;
        string first_half=s.substr(0,half);
        sort(first_half.begin(),first_half.end());
        string second_half=first_half;
        reverse(second_half.begin(),second_half.end());
        if(s.length()%2!=0)
        {
            char mid=s[half];
            return first_half+mid+second_half;
        }
        return first_half+second_half;
    }
};

File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.java
class Solution {
    public String smallestPalindrome(String s) {
        int half=s.length()/2;
        char[] halfchars=s.substring(0,half).toCharArray();
        Arrays.sort(halfchars);
        String first_half=new String(halfchars);
        String second_half=new StringBuilder(first_half).reverse().toString();
        if(s.length()%2!=0)
        {
            char mid=s.charAt(half);
            return first_half+mid+second_half;
        }
        return first_half+second_half;
    }
}